dnsmap: init at 0.36-unstable-2024-08-20

heywoodlh opened the pull request:
## Description of changes

Added dnsmap

Related: https://github.com/NixOS/nixpkgs/issues/81418

(I didn't add myself as a maintainer because of [all the other open commits I have in a pending state](https://github.com/NixOS/nixpkgs/pulls?q=author%3Aheywoodlh) -- please let me know if I should add the maintainer commit to this PR as well)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

heywoodlh committed:
dnsmap: init at 0.36-unstable-2024-08-20

superherointj commented on pkgs/by-name/dn/dnsmap/package.nix:
```diff
@@ -0,0 +1,35 @@
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "dnsmap";
+  version = "unstable-2024-06-16";
```
```suggestion
  version = "0.36-unstable-2024-06-16";
```

heywoodlh replied:
Resolved on latest commit

superherointj commented:
Eval is failing because of missing maintainer.

You could `cherry-pick` the commit that adds yourself as maintainer. And use same cherry-pick for all your PRs, it won't duplicate it. And on rebasing it will merge fine.

superherointj commented:
Since you are a maintainer now, you need to rebase this PR.

superherointj commented on pkgs/by-name/dn/dnsmap/package.nix:
```diff
@@ -0,0 +1,35 @@
+    rev = "b5b4b1a7764e141f2551584d9fad3a973951eafe";
+    hash = "sha256-RTZe0kb/Fq9kIdnHQO//OP+Uop2Z5r22Z7+rQdCFsl4=";
+  };
+
```
> strictDeps = true;

(Always have this for `stdenv.mkDerivation`. This should be the default.)

superherointj replied:
`preConfigure`, `configureFlags` is configure phase. Should precede buildInputs here.

superherointj commented on pkgs/by-name/dn/dnsmap/package.nix:
```diff
@@ -0,0 +1,35 @@
+
+  configureFlags = [ "--prefix=$(out)" ];
+
+  meta = {
```
meta.changelog is missing:

> changelog = "https://github.com/resurrecting-open-source-projects/dnsmap/releases/tag/${version}";

superherointj commented on pkgs/by-name/dn/dnsmap/package.nix:
```diff
@@ -0,0 +1,35 @@
+{ lib, stdenv, fetchFromGitHub, autoconf, automake }:
```
Use nixfmt (rfc).

heywoodlh commented:
@superherointj sorry for the delay on my end, I missed these notifications. I think I addressed all your feedback in the latest force push.

I wasn't sure if there was a better way to version dnsmap vs just `0.36-unstable-$DATE` -- because if you run dnsmap, you'll see the version is still `0.36` in the output -- but there have been commits to the repository since the `0.36` release. So I think my version naming clearly illustrates that the major version is `0.36` but there have been updates to it, therefore it's still "unstable". 

Please let me know if there's any other follow-up needed on my end

superherointj commented:
Please fix commit message:

> dnsmap: init at 0.36-unstable-2024-08-08

superherointj commented on pkgs/by-name/dn/dnsmap/package.nix:
```diff
@@ -0,0 +1,41 @@
+  configureFlags = [ "--prefix=$(out)" ];
+
+  buildInputs = [
+    autoconf
+    automake
```
Enable:
> strictDeps = true;

Good odds this is going to `nativeBuildInputs`.
Test cross compiling with `pkgsCross`.

> nix build .#pkgsCross.aarch64-multiplatform.dnsmap

superherointj commented on pkgs/by-name/dn/dnsmap/package.nix:
```diff
@@ -0,0 +1,41 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  autoconf,
+  automake,
+}:
```
```suggestion
{
  autoconf,
  automake,
  fetchFromGitHub,
  lib,
  stdenv,
}:
```
There isn't consensus on this, so this is an optional comment.
I favor alphabetical order because it is very objective.

superherointj commented on pkgs/by-name/dn/dnsmap/package.nix:
```diff
@@ -0,0 +1,41 @@
+  ];
+
+  meta = {
+    description = "Scan for subdomains using bruteforcing techniques ";
```
```suggestion
    description = "Scan for subdomains using brute-force techniques";
```

superherointj commented:
ping @heywoodlh

heywoodlh commented:
@superherointj thanks for the ping, will address the feedback you brought up sometime this week

heywoodlh commented:
@superherointj, sorry for the delay, I think I've addressed all your points on the latest force push.

superherointj commented:
> @superherointj, sorry for the delay, I think I've addressed all your points on the latest force push.

You did not.

heywoodlh commented:
@superherointj sorry! I forgot to commit and push that piece.
